### DISCUSSION TITLE
Opencore Always Resets RefindPlus Boot

### DISCUSSION THREAD

**Discussion by MarioG-X** [dakanji/RefindPlus#134] [General]
Just got OC to boot Ventura and it works fine. However, I have a bit of a headache and web searches were not fruitful. I have RP and Opencore installed in the same EFI, I just copied the OC directory from the USB to the EFI next to the Refind directory. When I boot into RP I see the OC boot entry fine, and if I select it, OC displays Ventura and that boots fine.

The problem is anytime I use the OC entry, for Ventura or Monterey, it resets booting from RP directly to OC, so the next time the machine boots it's into OC, not RP so no Linux. I looked for something that would stop this but found nothing. I did find on the OC site a mention that in a recent update this was done to prevent something else from booting! Not Good! This is obviously a royal pain since to fix it I have to boot to recovery to install Refind, then boot again to replace Refind with RP, then boot again to what I need. If this can't be resolved I will forgo Ventura. BTW, I also looked at mybootmgr which is way over my head. I was hoping OC had an option to not force itself each time it was booted. Thanks for any help.

see:
https://dortania.github.io/OpenCore-Post-Install/multiboot/bootstrap.html#prerequisites

If everything is set up correctly, the first boot will have OpenCore create a new boot option in our BIOS (pointing to EFI/OC/OpenCore.efi) and future boots will update the entry making sure it's correct and ensuring it is still present. This now allows us to remove BOOTx64.efi and not worry about other OSes overwriting the EFI/BOOT/BOOTx64.efi path.

**Comment by startergo** [comment]
>Once you boot OpenCore for the first time, LauncherOption will install itself as the top boot priority, making OpenCore always launch. Combined with RequestBootVar, all boot options must go through OpenCore, ensuring seamless usage even with OS installation and updates.

Have you tried with ```RequestBootVar``` disabled?

https://github.com/dortania/OpenCore-Legacy-Patcher/blob/main/docs/TROUBLESHOOTING.md

Also in the config file of OC make sure that you set delete on all Nvram options.

I use chain loading on OC from rEFInd on a PC and I use 0.8.7 OC version and there is no hijacking.

I disable ```LauncherOption```:
https://github.com/startergo/x299-WS-PRO-SE-OpenCore/blob/main/Config%20Plist/OpenCore%200.8.7%20dev/config.plist#L889-L890

**Comment by dakanji** [comment]
> The problem is anytime I use the OC entry, for Ventura or Monterey, it resets booting from RP directly to OC

That is an OpenCore Boot Coup. Go to https://forums.macrumors.com/threads/2231693 and look for how to deal with such under `Hints and Tips`.

>  BTW, I also looked at MyBootMgr which is way over my head

I have heard that a few times and always puzzled about what is so difficult about downloading and running a couple of apps or following an itemised list of actions.

```
- Execute a System Reset to ensure dodgy stuff (such as your OCLP hijack) is removed
- Download and run MyBootMgrSetup to place the apps ready for use
- Run the EnterRecovery App to disable SIP if on El Capitan or newer
- Run the ConfigFactory App to configure and stage the setup
- Run the DeployConfig App to move the setup to a designated ESP
- Run the BootBlesser App using `Soft Bless` to safely test the setup
- Run the SwitchRunMode App to change from `DBG` to `REL`
- Run the BootBlesser App using `Firm Bless` to finalise the setup
```

Everything else on the page is background information as I have put virtually everything a user is likely to need on the page so that they don't need to rely on me to resolve most issues but I suppose people simply just don't like reading. They take one look at the stuff written and run away. I suppose I also quite prefer that at the end of the day as it filters a lot of problems out right off the bat.

Bottomline is that if you were using MyBootMgr, put in 10 mins to read what is written and followed the instructions on absorbing third party configured OpenCore Ventura when running the `ConfigFactory` app you wouldn't have had this issue. Even if you did, the answer is sitting right there.

MyBootMgr support is only provided on when used on Classic MacPro but it can be used with other units. You just need to replace the OpenCore it creates with yours in such cases.

Anyway, you can just go to the hints and tips bit there, look for the boot coup items and sort the issue out.

**Comment by MarioG-X** [comment]
Thanks, I will try some suggestions and let you know if I can get it working.

dakanji, 10 minutes? I have read that thread several times and spent hours studying it  multiple times over the last month. I guess I am just one of the people that don’t get it.   Btw, the machine is a 2014 Mac mini.

**Comment by dakanji** [comment]
> I guess I am just one of the people that don’t get it. Btw, the machine is a 2014 Mac mini

I agree that some don't get it indeed. I have found that the main reason for this is that people can struggle to simply follow the instructions and forget all sorts of other stuff they have come across that are swirling about in their minds:

```
- Execute a System Reset to ensure dodgy stuff (such as OCLP hijacks) are removed
- Download and run MyBootMgrSetup to place the apps ready for use
- Run the EnterRecovery App to disable SIP if on El Capitan or newer
- Run the ConfigFactory App to configure and stage the setup
- Run the DeployConfig App to move the setup to a designated ESP
- Run the BootBlesser App using `Soft Bless` to safely test the setup
- Run the SwitchRunMode App to change from `DBG` to `REL`
- Run the BootBlesser App using `Firm Bless` to finalise the setup
```

I suppose it is a bit more difficult when not on a Classic MacPro. In such cases, especially for `Ventura`, it is probably best to just stick to resolving the boot coup and disabling `LauncherOption` to prevent it in future. You don't need to touch `RequestBootVarRouting`.

**Comment by MarioG-X** [comment]
I got it working fine on an iMac and Mini by setting `RequestBootVarRouting` to `false` and `LauncherOption` to `Disabled`. Both machines now boot Ventura and do not set booter to OC. Took 5 minutes. THANKS GUYS! BTW, I did find what I _think_ is a Refind or RP bug.

BTW, startergo, you mentioned to make sure NVRAM was off but I only found two NVRAM options both were off by default.

**Comment by startergo** [comment]
I was referring to the add/delete sections for the NVRAM. If you disable RequestBootVarRouting
how do you set the default boot option in OC?
